packages/waku/src/router/client.ts

```diff
@@ -805,27 +826,28 @@ const InnerRouter = ({ initialRoute }: { initialRoute: RouteProps }) => {
 
   useEffect(() => {
     const callback = (path: string, query: string) => {
-      const fn = () => {
+      const fn = async () => {
         const url = new URL(window.location.href);
         url.pathname = path;
         url.search = query;
         url.hash = '';
-        if (path !== '/404') {
-          window.history.pushState(
-            {
-              ...window.history.state,
-              waku_new_path: url.pathname !== window.location.pathname,
-            },
-            '',
-            url,
-          );
+        try {
+          await changeRoute(parseRoute(url), {
+            skipRefetch: true,
+            shouldScroll: false,
+          });
+        } finally {
+          if (path !== '/404') {
+            window.history.pushState(
+              {
+                ...window.history.state,
+                waku_new_path: url.pathname !== window.location.pathname,
+              },
+              '',
+              url,
+            );
+          }
         }
```

Comment:
why not use `.finally` here like the other use?

Reply:
Good question. I should have not switched it to an async def. I'll fix.

Reply:
Oh... it's so the error can fall through. I think we need it.